### PR TITLE
Added Descope + Expo Sample App

### PR DESCRIPTION
**Pull request by gaokevin1**
This is an example of how you would use Descope as an OAuth provider with Expo and AuthSession.

### CODE REVIEW

**Comment by gaokevin1** [comment]
@amandeepmittal Could you please review and merge this for me? Thanks!

**Comment by amandeepmittal** [comment]
Hi @gaokevin1, I'll take a look at this PR. Thanks for opening it and apologies it completely slipped me.

**Comment by gaokevin1** [comment]
@amandeepmittal Any updates or thoughts here?

**Comment by amandeepmittal** [comment]
Apologies once again @gaokevin1, I'll take look at it this week.

File: with-descope/package.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+  "scripts": {
+    "start": "expo start",
+    "android": "expo run:android",
+    "ios": "expo run:ios",
+    "web": "expo start --web"
+  },
+  "dependencies": {
+    "expo": "~49.0.15",
```
Comment (amandeepmittal) [inline with-descope/package.json:9]: We recently released [SDK 50](https://expo.dev/changelog/2024/01-18-sdk-50), can you please upgrade it to that?

File: with-descope/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+- Install with `yarn` or `npm install`.
+- Create your own project on [Descope](https://www.descope.com/sign-up).
+- Open `App.js` and replace `descopeProjectId` with your [Descope Project ID](#descope-project-id).
+- Run `yarn start` or `npm run start` to try it out.
+
+#### AuthSession callback URL
+
+The AuthSession helps you with browser authentication, without the need of an additional server or website. To use this with Auth0 authentication flows, we need to tell Auth0 that the callback URLs are allowed.
+
+Each Expo user has it's own URL for different projects, the basic structure of this URL is `https://auth.expo.io/@your-username/your-expo-app-slug`. If you are signed in as `awesome-ppl`, and your app is called `meme-explorer`, your URL looks like `https://auth.expo.io/@awesome-ppl/meme-explorer`.
```
Comment (amandeepmittal) [inline with-descope/README.md:23]: We have deprecated this in a previous SDK release and I'd avoid recommending it: https://docs.expo.dev/versions/latest/sdk/auth-session/#authsessiongetredirecturlpath.

**Comment by gaokevin1** [comment]
@amandeepmittal sorry I forgot to check on this, I fixed the issues you reported.

**Review by amandeepmittal — APPROVED** [review]
Thank you for updating the example!

**Pull request merged**